src/main/java/duke/system/Parser.java

```diff
@@ -0,0 +1,172 @@
+     */
+    public Parser(String in) {
+        String tempDate = null;
+        String tempCommand = "";
+        String[] result = in.split("\\s");
+        String tempArg = "";
+        try {
+            if (result[0].equals("done") || result[0].equals("find")) {
+                tempCommand = result[0];
+                if (result.length <= 1) {
+                    throw new DukeException.NoDescriptionException(result[0]);
+                } else {
+                    tempArg = result[1];
+                }
+            } else if (result[0].equals("todo") || result[0].equals("delete")) {
+                String temp = in.substring(in.indexOf(" ") + 1);
+                tempArg = temp;
+                if (temp.equals("todo") || temp.equals("delete")) {
+                    throw new DukeException.NoDescriptionException(result[0]);
+                } else {
+                    tempCommand = result[0];
+                }
+            } else if (result[0].equals("deadline") || result[0].equals("event")) {
+                String firstParam = in.substring(in.indexOf("/") + 1);
+                if (firstParam.equals("deadline") || firstParam.equals("event")) {
+                    throw new DukeException.NoDescriptionException(result[0]);
+                } else {
+                    int dateIndex = Math.max(firstParam.indexOf("by "), firstParam.indexOf("at "));
+                    if (dateIndex == -1) {
+                        throw new DukeException.NoDescriptionException(result[0]);
+                    } else {
+                        tempCommand = result[0];
+                        tempDate = firstParam.substring(dateIndex + 3);
+                        firstParam = in.substring(in.indexOf(" ") + 1);
+                        tempArg = firstParam.substring(0, firstParam.indexOf("/") - 1);
+                    }
+                }
+            } else {
+                try {
+                    tempCommand = String.valueOf(predefinedCommand.valueOf(result[0]));
+                } catch (IllegalArgumentException ex) {
+                    throw new DukeException.UnknownCommandException();
+                }
+            }
+        } catch (DukeException ex) {
+            tempCommand = "error";
+            tempArg = ex.getMessage();
+        }
+        this.command = tempCommand;
+        this.argument = tempArg;
+        this.date = tempDate;
+    }
+
+    /**
+     * Getter method
+     * @return the private variable command
+     */
+    public String getCommand() {
+        return this.command;
+    }
+
+    /**
+     * Getter method
+     * @return the private variable argument
+     */
+    public String getDate() {
+        return this.argument;
+    }
+
+    /**
+     *
+     * @param inputList - take in the list and do corresponding action according to the command
+     * @return a string to be printed to the console by UI
+     */
+    public String print(TaskList inputList) {
```

Comment:
This method is called 'print' but in fact it is adding certain new task to the task list. Perhaps you can try to refactor this method to make it cleaner! :)